docs/source/federation-spec.mdx

```diff
@@ -0,0 +1,625 @@
+The `for` argument describes the purpose of a `@link`. Currently accepted values are `SECURITY` or `EXECUTION`. Core schema-aware servers such as Apollo Router and Gateway will refuse to operate on schemas that contain `@link`s to unsupported specs which are `for: SECURITY` or `for: EXECUTION`.
+
+By default, `@link`ed definitions will be namespaced, i.e., `@federation__requires`. The `as` argument lets you pick the name for this namespace:
+```graphql
+extend schema @link(url: "https://specs.apollo.dev/federation/v2.0", as: "fed2")
+type User {
+  metrics: Metrics @external
+  favorites: UserFavorites @fed2__requires(fields: "metrics")
+}
+```
+
+The `import` argument enables you to import external definitions into your namespace, so they are not prefixed.
+
+```graphql
+    schema
+      @link(url: "https://specs.apollo.dev/link/v1.0")
+      @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key", "@requires", "@provides", "@external", { name: "@tag", as: "@mytag" }, "@extends", "@shareable", "@inaccessible", "@override"])
+```
+
+In the example above, we import various directives from `federation/v2.0` into our namespace. We also rename one of them, bringing in federation's `@tag` as `@mytag` to distinguish it from a different `@tag` directive already in the schema.
+
+### `@override`
+
+```graphql
+directive @override(from: String!) on FIELD_DEFINITION
+```
+
+The `@override` directive indicates that the current subgraph takes responsibility for resolving a field _away_ from the subgraph specified in the `from` argument.
+
+> Only one subgraph can `@override` any given field. If multiple subgraphs attempt to `@override` the same field, a composition error occurs.
+
+In the following example, the `User.name` field is resolved by Subgraph B _instead of_ Subgraph A.
+
+<CodeColumns>
+
+```graphql title="Subgraph A"
+type User @key(fields: "id") {
+  id: ID!
+  name: String
+}
+```
+
+```graphql title="Subgraph B"
+type User @key(fields: "id") {
+  id: ID!
+  name: String @override(from: "SubgraphA")
+}
+```
+
+</CodeColumns>
+
+### `@provides`
+
+```graphql
+directive @provides(fields: FieldSet!) on FIELD_DEFINITION
+```
+
+The `@provides` directive declares a set of entity fields that a subgraph _can_ resolve, but _only_ at a particular schema path. This directive is used whenever a subgraph _can't_ resolve those entity fields at _other_ paths.
+
+Consider this example subgraph schema:
+
+```graphql
+type Review @key(fields: "id") {
+  product: Product @provides(fields: "name")
+}
+
+extend type Product @key(fields: "upc") {
+  upc: String @external
+  name: String @external
+}
+```
+
+By default, this subgraph _can't_ resolve the `Product.upc` or `Product.name` fields, so it marks those fields as `@external`. However, this subgraph _can_ resolve the `Product.name` field specifically when it's a subfield of `Review.product`. So, it applies the `@provides` directive to `Review.product` to indicate this fact to the graph router.
```

Comment:
technically federation v2 does not require `@external` on keys anymore, similarly if that is the only query path that `Product.name` can be resolved then there is no need for `@external` neither

Reply:
Good call, this content is now cut so no problem here